picard/album.py

```diff
@@ -285,6 +303,42 @@ def _release_request_finished(self, document, http, error):
             if parsed or error:
                 self._finalize_loading(error)
 
+    def _request_recording_relationships(self, offset=0, limit=RECORDING_QUERY_LIMIT, config=None):
+        if not config:
+            config = get_config()
+        inc = {
+            'artist-rels',
+            'recording-rels',
+            'release-rels',
+            'url-rels',
+            'work-rels'
+        }
+        log.debug('Loading recording relationships for %r (offset=%i, limit=%i)' % (self, offset, limit))
+        self._requests += 1
+        self.load_task = self.tagger.mb_api.browse_recordings(
+            self._recordings_request_finished,
+            inc=tuple(inc),
+            release=self.id,
+            limit=limit,
+            offset=offset
+        )
+
+    def _recordings_request_finished(self, document, http, error):
+        self._requests -= 1
+        if error:
+            self.error_append(http.errorString())
+            self._finalize_loading(error)
+        else:
+            for recording in document.get('recordings', []):
+                self._recordings_map[recording.get('id')] = recording
```

Comment:
If `recording.get('id')` is ensured to always work, we can write `recording['id']` and if it doesn't, I'm not sure we want to store the value for the `None` key. Since those come the web service, I'd rather ensure we have an `id`:

```python
for recording in document.get('recordings', []):
    recording_id = recording.get('id')
    if recording_id:
        self._recordings_map[recording_id] = recording
```

What do you think?

Reply:
Yes, I changed this and rebased